zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -103,6 +104,7 @@
    */
   private final Map<String, InterpreterSetting> interpreterSettings =
       Maps.newConcurrentMap();
+  private final ReentrantReadWriteLock interpreterSettingsLock = new ReentrantReadWriteLock();
```

Comment:
Actually interpreterSettings is ConcurrentHashMap, I think we don't need extra lock here.

Reply:
Yes, I thought it and wanted to confirm it. Then, I'll remove all of the locks related to interpreterSettings in this class.

Reply:
Fixed